packages/web/utils/background-routes/index.ts

```diff
@@ -0,0 +1,166 @@
+import {
+  OptimizedRoutesParams,
+  RouteWithInAmount,
+  SplitTokenInQuote,
+  Token,
+  TokenOutGivenInRouter,
+} from "@osmosis-labs/pools";
+
+import {
+  checkResponseAndDecodeError,
+  decodeRouteWithInAmount,
+  decodeSplitTokenInQuote,
+  encodeCalculateTokenOutByTokenInParameters,
+  encodeGetOptimizedRoutesByTokenInParameters,
+  encodeOptimizedRoutesParams,
+  encodeRouteByTokenInParameters,
+} from "./coding";
+import {
+  EncodedRequest,
+  EncodedResponse,
+  OptimizedRoutesWorker,
+} from "./worker";
+
+/** Router that delegates search problem to background thread, useful for unblocking the main thread for UI/DOM updates. */
+export class BackgroundRoutes implements TokenOutGivenInRouter {
+  protected static singletonWorker: OptimizedRoutesWorker | null = null;
```

Comment:
Why is this a static variable instead of an instance variable?

Reply:
Hmm I think my intention was to always share a single router thread, since then we'd share the cache in the router.

Could be a large assumption though, thoughts?

Reply:
Theoretically, if there were multiple instances of BackgroundRoutes, there could be an issue since they would all share the same Worker. Seems like an opportunity for a bug to me. I think it makes more sense to make it an instance variable.

Reply:
This is not obvious, so I'll add a readme to explain this.